src/Refactoring-UI/ReExtractToTemporaryDriver.class.st

```diff
@@ -0,0 +1,136 @@
+Class {
+	#name : 'ReExtractToTemporaryDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'sourceInterval',
+		'newVariableName',
+		'selector',
+		'class',
+		'shouldEscape'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> changes [
+	
+	^ refactoring privateTransform; changes
+
+]
+
+{ #category : 'resources' }
+ReExtractToTemporaryDriver >> configureRefactoring [
+
+	refactoring := RBExtractToTemporaryRefactoring 
+							extract: sourceInterval
+						 	to: newVariableName
+							from: selector
+							in: class
+]
+
+{ #category : 'instance creation' }
+ReExtractToTemporaryDriver >> extract: anInterval to: aString from: aSelector in: aClass [
+
+	class := RBNamespace new classObjectFor: aClass.
+	selector := aSelector.
+	sourceInterval := anInterval.
+	newVariableName := aString
+]
+
+{ #category : 'initialization' }
+ReExtractToTemporaryDriver >> initialize [ 
+	
+	super initialize.
+	shouldEscape := false.
+]
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> requestSelector [
+
+	^ self requestUserInput: 'Please provide a selector'
+			 label: 'The selector should be defined.'  
+			 validate: [ :string :presenter | self validateSelector: string onPresenter: presenter ]
+]
```

Comment:
Two things here, both are non-blocking:
- since Driver should only be called through UI, chances that selector is invalid are close to 0, so this check is optional in the driver and will always pass
- correct way of asking user for selector would be to use selection presenter that you fill with available selectors from the class, but again this is not needed since this will be correctly set always

This check is important when we are invoking this refactoring programatically, and it should then raise an exception, and not ask for new selector.

My suggestion might be to remove this code from driver. Would like to hear you thoughts as well @Ducasse